mungegithub/mungers/milestone-maintainer.go

```diff
@@ -324,12 +318,25 @@ func labelsIncompleteState(issue *githubapi.Issue, notification *c.Notification,
 		return nil
 	}
 
+	if warningIsCurrent(notification, *message, commentCreatedAt, warningInterval) {
+		return nil
+	}
+
 	return &milestoneNotification{
 		description: milestoneLabelsIncomplete,
 		message:     *message,
 	}
 }
 
+// warningIsCurrent indicates whether the given notification
+// represents is an up-to-date warning.  The notification is the
+// previous warning from the munger.  The message is the warning
+// generated from the current labels on the issue.
+func warningIsCurrent(notification *c.Notification, message string, commentCreatedAt *time.Time, warningInterval time.Duration) bool {
+	hasNotification := (notification != nil && notification.Arguments == milestoneLabelsIncomplete)
+	return hasNotification && notification.Context == message && time.Since(*commentCreatedAt) < warningInterval
```

Comment:
Check that `commentCreatedAt` is not nil.

Reply:
It's not necessary in this case.  If notification is not nil, comment is not nil, and commentCreatedAt will not be nil.

Reply:
Sorry, I don't follow. `c.ParseNotification` can return a non-nil notification even if the comment it is created from has a nil value for `CreatedAt`. So {notification not nil} does imply {comment not nil} but does not imply {commentCreatedAt not nil}.

Reply:
Done